apiserver/README.md

```diff
@@ -119,744 +157,1841 @@ nodes:
+    helm -n ray-system install kuberay-apiserver kuberay/kuberay-apiserver -n ray-system --create-namespace
+    ```
 
-```
-curl localhost:31888
-{"code":5, "message":"Not Found"}
-```
+3. The APIServer expose service using `NodePort` by default. You can test access by your host and port, the default port is set to `31888`. The examples below assume a kind (localhost) deployment. If Kuberay API server is deployed on another type of cluster, you'll need to adjust the hostname to match your environment.
 
-3. You can create `RayCluster`, `RayJobs` or `RayService` by dialing the endpoints. The following is a simple example for creating the `RayService` object, follow [swagger support](https://ray-project.github.io/kuberay/components/apiserver/#swagger-support) to get the complete definitions of APIs.
-
-```shell
-curl -X POST 'localhost:31888/apis/v1alpha2/namespaces/ray-system/compute_templates' \
---header 'Content-Type: application/json' \
---data '{
-  "name": "default-template",
-  "namespace": "ray-system",
-  "cpu": 2,
-  "memory": 4
-}'
-
-curl -X POST 'localhost:31888/apis/v1alpha2/namespaces/ray-system/services' \
---header 'Content-Type: application/json' \
---data '{
-  "name": "user-test-1",
-  "namespace": "ray-system",
-  "user": "user",
-  "serveDeploymentGraphSpec": {
-      "importPath": "fruit.deployment_graph",
-      "runtimeEnv": "working_dir: \"https://github.com/ray-project/test_dag/archive/c620251044717ace0a4c19d766d43c5099af8a77.zip\"\n",
-      "serveConfigs": [
-      {
-        "deploymentName": "OrangeStand",
-        "replicas": 1,
-        "userConfig": "price: 2",
-        "actorOptions": {
-          "cpusPerActor": 0.1
-        }
-      },
-      {
-        "deploymentName": "PearStand",
-        "replicas": 1,
-        "userConfig": "price: 1",
-        "actorOptions": {
-          "cpusPerActor": 0.1
-        }
+    ```sh
+    curl localhost:31888
+    ...
+    {
+      "code": 5,
+      "message": "Not Found"
+    }
+    ```
+
+4. You can create `RayCluster`, `RayJobs` or `RayService` by dialing the endpoints. The following is a simple example for creating the `RayService` object, follow [swagger support](https://ray-project.github.io/kuberay/components/apiserver/#swagger-support) to get the complete definitions of APIs.
+
+    Please note that:
+
+    * The following examples use the `ray-system` namespace. If not already created by using the helm install steps above, you can create it prior to executing the curl examples by running `kubectl create namespace ray-system`
+    * The examples assume that the cluster has at least 2 CPUs available and 4 GB of free memory. You can either increase the CPUs available to your cluster (docker settings) or reduce the CPU request in the `compute_templates` request.
+    * If you are running the service and the kuberay operator on Apple Silicon Machine, you might want to use the `rayproject/ray:2.7.0-aarch64`image.
+
+    ```sh
+    # Create a template
```

Comment:
Maybe explain what a template is and what it does

Reply:
The other thing that I was thinking about is more cluster creation examples:
1. HA cluster
2. Using volumes (different volumes types)
3. Creating an Ingress

Reply:
Will expand the explanation on compute templates. I think tha the other topics are for later PR(s). 
tedhtchang has #1408. Do you mind opening tickets for each item you have listed here (or others you can thin of)?

I rather iterate over multiple PR(s) on documentation improvements.

Reply:
Agree with the approach. Approve